webapp/tbapi/tbapi/blueprints/api.py

```diff
@@ -1,20 +1,138 @@
-from flask import Blueprint, render_template, make_response, jsonify, flash, redirect, url_for, request, current_app
+from flask import Flask, Blueprint, render_template, make_response, jsonify, flash, redirect, url_for, request, current_app, abort
 from tbapi.models.mpower import Patient, User
+from flask_swagger import swagger
+from tbapi.models.tbapp import Note
 
 bp = Blueprint('api', __name__)
 
-@bp.route('/patients')
-def patient_index():
-    return jsonify(patients=[i.serialize for i in Patienst.query.all()])
+@bp.route('/notes/<int:id>', methods=['GET']) # note id
+def getNote(id):
+    if id: # note id
+        return jsonify(notes=[i.serialize for i in Note.get_by_note(id)])
+    else:
+        abort(400, 'note id required')
+        
 
-@bp.route('/patients/<int:id>')
-def patient(id):
-    return jsonify(Patient.query.filter_by(id=id).first().serialize)
+@bp.route('/notes/', methods=['GET']) # patient_id
+def getPatient():
+    try:
+        patient_id = int(request.args.get('patient_id'))
+    except:
+        abort(400, 'patient_id required')
+   
+    return jsonify(notes=[i.serialize for i in Note.get_by_patient_id(patient_id)])
+    
+    """Access basics for patient notes
 
-@bp.route('/users')
-def user_index():
-    return jsonify(users=[i.serialize for i in User.query.all()])
+    1. GET /api/notes/?patient_id=int  returns all notes by specified patient_id
+    2. GET /api/notes/{int:id} returns note specified by id.
 
-@bp.route('/users/<int:id>')
-def user(id):
-    return jsonify(User.query.filter_by(id=id).first().serialize)
+    returns patient notes id, patient_id, text, author_id, created, lastmod in JSON
+    ---
+    tags:
+        - Note GET
+    operationId: getNotes
+    parameters:
+      - in: body
+        name: body
+        schema:
+          id: note
+          properties:
+              id:
+                  type: integer
+                  format: int64
+                  description: id for note
+              patient_id:
+                  type: integer
+                  description: patient_id
+              text:
+                  type: string
+                  description: body of note
+              author_id:
+                  type: integer
+                  description: author_id
+              created:
+                  type: string
+                  format: date-time
+                  description: created
+              lastmod:
+                  type: string
+                  format: date-time
+                  description: lastmod
+    produces:
+        - application/json
+    responses:
+        200:
+          description:
+              "Returns {notes}"
+    """
+    
+@bp.route('/notes', methods=['POST'])
```

Comment:
Whoa, hold the phone. This line might need the trailing `/` after all.

I'm baffled. When I try to send a `POST` to `/api/v1.0/notes`, the CORS preflight request fails with a `301 MOVED PERMANENTLY`.
<img width="582" alt="tb_asistente_diario" src="https://user-images.githubusercontent.com/30454698/46762724-eba61380-cca5-11e8-8e21-c838e9647ead.png">
<img width="320" alt="tb_asistente_diario" src="https://user-images.githubusercontent.com/30454698/46762579-73d7e900-cca5-11e8-967a-a5e7c1060f71.png">



When I change it to `/api/v1.0/notes/` the preflight request succeeds, but the main `POST` request fails:
<img width="582" alt="tb_asistente_diario" src="https://user-images.githubusercontent.com/30454698/46762695-d4ffbc80-cca5-11e8-9906-7a31dfe4c519.png">
<img width="320" alt="tb_asistente_diario" src="https://user-images.githubusercontent.com/30454698/46762640-a2ee5a80-cca5-11e8-848e-e78243ec9a61.png">
<img width="320" alt="tb_asistente_diario" src="https://user-images.githubusercontent.com/30454698/46762657-b13c7680-cca5-11e8-9379-0fefaa2ce703.png">


The only way I've been able to get this to work is by changing this line to:

```python
@bp.route('/notes/', methods=['POST'])
```
<img width="582" alt="tb_asistente_diario" src="https://user-images.githubusercontent.com/30454698/46762831-2f991880-cca6-11e8-95f2-dd146868cd96.png">

Reply:
CC @ivan-c ^, the CORS mystery deepens...

Reply:
It turns out that Flask-CORS was getting mixed up by having one route (`GET`) with a trailing slash, and one route (`POST`) without it.

Either adding a trailing slash, or removing a trailing slash from both routes solves our problem. For simplicity, we're going to remove them.